packages/starrating/src/react/index.js

```diff
@@ -0,0 +1,155 @@
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const interactive = !!nextProps.onChange
+    if (interactive === this.props.interactive) return
+
+    this.setState({ interactive })
+  }
+
+  handleInputChanged(event) {
+    if (!this.state.interactive) return
+    this.props.onChange(event.target.value, event)
+  }
+
+  handleStarClicked(index, event) {
+    if (!this.state.interactive) return
+    this.props.onChange(index + 1, event)
+  }
+
+  handleStarEnter(index, event) {
+    if (!this.state.interactive) return
+    this.setState(() => ({ hoverIndex: index }))
+  }
+
+  handleStarLeave(index, event) {
+    if (!this.state.interactive) return
+    this.setState(() => ({ hoverIndex: null }))
+  }
+
+  get stars() {
+    const { hoverIndex, interactive } = this.state
+    const { starCount, value } = this.props
+
+    const hasValidRating = !Number.isNaN(parseFloat(value, 10))
+    const isHovering = hoverIndex !== null
+
+    const valueRounded = Math.floor(value * 2) / 2
+    const fullStars = Math.floor(valueRounded)
+    const halfStars = Math.floor(valueRounded) !== valueRounded
+
+    return [...Array(starCount)].map((_, index) => {
+      let active = false
+      let appearance
+
+      if (isHovering) {
+        if (index <= hoverIndex) {
+          active = true
+          appearance = Star.appearances.full
+        } else {
+          appearance = Star.appearances.empty
+        }
+      } else {
+        if (interactive && !hasValidRating) {
+          appearance = Star.appearances.empty
+        } else if (index < fullStars) {
+          active = true
+          appearance = Star.appearances.full
+        } else if (index === fullStars && halfStars) {
+          active = true
+          appearance = Star.appearances.half
+        } else {
+          appearance = Star.appearances.full
+        }
+      }
+
+      return (
+        <Star
+          active={active}
+          appearance={appearance}
+          index={index}
+          interactive={interactive}
+          key={index}
+          onClick={this.handleStarClicked}
+          onEnter={this.handleStarEnter}
+          onLeave={this.handleStarLeave}
+        />
+      )
+    })
+  }
+
+  render() {
+    const { interactive } = this.state
+    const { starCount, value } = this.props
+
+    return (
+      <div {...styles.starRating(this.props)}>
+        {interactive && (
+          <label>
+            <ScreenReaderText>Rate</ScreenReaderText>
+            <ScreenReaderInput
+              type="range"
+              onChange={this.handleInputChanged}
+              min={1}
+              max={starCount}
+              step={1}
+              value={value || 0}
+            />
+          </label>
+        )}
+
+        {this.stars}
+
+        <ScreenReaderText>This is rated {value}</ScreenReaderText>
+      </div>
+    )
+  }
+}
+
+StarRating.propTypes = {
+  onChange: PropTypes.func,
+  starCount: PropTypes.number,
```

Comment:
fancy!  not strictly a requirement.  Should we keep?

Reply:
i wouldn't document it as an option but I would define the propType is this case. the other idea would be to not make it prop and define elsewhere. this seemed easy enough but i could go either way

Reply:
I'm fine for the moment.  Don't worry about it.  I've had enough experiences adding easy things that were required only to have problems and support effort required to maintain them.  There's also the need to work around these things in the future as we read and code in this component. It's also way easier to add things as they're needed vs retiring them. 
So I try to be careful not to over deliver on things not specifically needed so I can ensure more time for what is known to be needed.  But, you know, sometimes there are extra features we add that end up being unexpectedly used and successful, so it's hard to say "never".